src/ControlSystem/Systems/Shape.hpp

```diff
@@ -108,13 +115,37 @@ struct Shape : tt::ConformsTo<protocols::ControlSystem> {
         Parallel::GlobalCache<Metavariables>& cache,
         const LinkedMessageId<double>& measurement_id) {
       auto& control_sys_proxy = Parallel::get_parallel_component<
-          ControlComponent<Metavariables, Shape<Horizon, DerivOrder>>>(cache);
+          ControlComponent<Metavariables, Shape>>(cache);
 
       Parallel::simple_action<::Actions::UpdateMessageQueue<
           QueueTags::Strahlkorper<Frame::Distorted>, MeasurementQueue,
           UpdateControlSystem<Shape>>>(control_sys_proxy, measurement_id,
                                        strahlkorper);
     }
+
+    template <::domain::ObjectLabel MeasureHorizon, typename Metavariables>
+    static void apply(
+        measurements::BothHorizons::FindHorizon<MeasureHorizon> /*meta*/,
+        const Strahlkorper<Frame::Distorted>& strahlkorper,
+        Parallel::GlobalCache<Metavariables>& cache,
+        const LinkedMessageId<double>& measurement_id) {
+      // The measurement event will call this for both horizons, but we only
+      // need one of the horizons. So if it is called for the wrong horizon,
+      // just do nothing.
+      if constexpr (MeasureHorizon == Horizon) {
+        auto& control_sys_proxy = Parallel::get_parallel_component<
+            ControlComponent<Metavariables, Shape>>(cache);
+
+        Parallel::simple_action<::Actions::UpdateMessageQueue<
+            QueueTags::Strahlkorper<Frame::Grid>, MeasurementQueue,
```

Comment:
`Frame::Distorted`?  That's whet everything else in these functions uses, at least.